pkgs/misc/cups/drivers/mfcj430wcupswrapper/default.nix

```
@@ -0,0 +1,91 @@
+## Brother MFC-J430W CUPS wrapper driver
+#
+# This compiles the official Brother CUPS wrapper sources.
+#
+#
+## Install the driver
+# Using the standard method as indicated in
+#  https://nixos.wiki/wiki/Printing
+# by adding the following to the configuration.nix:
+#
+# {
+#   ...
+#   services.printing.enable = true;
+#   services.printing.drivers = [ mfcj430w-cupswrapper ];
+#   ...
+# }
+#
+#
+## Setup
+#     URI example
+#     ~  `lpd://BRW0080927AFBCE/binary_p1`
+#     ~  `ipp://your-printer-ip-address:631/ipp`
+#
+#
+## Testing
+# . used on NixOS 20.09 with Cinnamon using system-config-printer
+#
+
+
+{ lib, stdenv, fetchurl, makeWrapper, cups, perl, coreutils, gnused, gnugrep, pkgs, mfcj430wlpr }:
+
+stdenv.mkDerivation rec {
+  pname = "mfcj430w-cupswrapper";
+  version = "3.0.0-1";
+
+  src = fetchurl {
+    url = "https://download.brother.com/welcome/dlf006803/mfcj430wcupswrapper-src-${version}.tar.gz";
+    sha256 = "1gryrjx5cbfzrzf6qvbbqnprs3zrkmzvrrbggjy09f8apa2c5dlb";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ cups perl coreutils gnused gnugrep mfcj430wlpr ];
+
+  patchPhase = ''
+    WRAPPER=cupswrapper/cupswrappermfcj430w
+
+    substituteInPlace $WRAPPER \
+    --replace /opt "${mfcj430wlpr}/opt" \
+    --replace /usr "${mfcj430wlpr}/usr" \
+    --replace /etc "$out/etc"
+
+    substituteInPlace $WRAPPER \
+    --replace "\`cp " "\`cp -p " \
+    --replace "\`mv " "\`cp -p "
+  '';
+
+  buildPhase = ''
+    cd brcupsconfig
+    make all
+    cd ..
+  '';
+
+  installPhase = ''
+    TARGETFOLDER=$out/opt/brother/Printers/mfcj430w/cupswrapper/
+    CUPSPPD_DIR=$out/share/cups/model
+    CUPSFILTER_DIR=$out/lib/cups/filter
+
+    mkdir -p $TARGETFOLDER
+
+    cp brcupsconfig/brcupsconfpt1 $TARGETFOLDER
+    cp cupswrapper/cupswrappermfcj430w $TARGETFOLDER/
+    cp ppd/brother_mfcj430w_printer_en.ppd $TARGETFOLDER/
+
+    mkdir -p $CUPSPPD_DIR
+    ln -s $TARGETFOLDER/brother_mfcj430w_printer_en.ppd $CUPSPPD_DIR
```

```suggestion
    mkdir -p $out/opt/brother/Printers/mfcj430w/cupswrapper/ $out/share/cups/model
    cp brcupsconfig/brcupsconfpt1 cupswrapper/cupswrappermfcj430w ppd/brother_mfcj430w_printer_en.ppd $out/opt/brother/Printers/mfcj430w/cupswrapper/
    
    ln -s $TARGETFOLDER/brother_mfcj430w_printer_en.ppd $out/lib/cups/filter
```
Do we really need $out/opt?